src/opa.js

```diff
@@ -172,9 +251,12 @@ class LoadedPolicy {
     this.dataHeapTop = this.baseHeapTop;
   }
 
-  // evaluate will evaluate the loaded policy with the given input and
-  // return the result set. This should be re-used for multiple evaluations
-  // of the same policy with different inputs.
+  /**
+   * Evaluates the loaded policy with the given input and
+   * return the result set. This should be re-used for multiple evaluations
+   * of the same policy with different inputs.
+   * @param {unknown} input
```

Comment:
Should `input` be an `object` type?

Reply:
`unknown` is basically a fancy way of saying anything could be returned to this, cast it before you try using it. I can make it `object` if you want to make it simpler though.